malcolm/modules/scanning/parts/directorymonitorpart.py

```diff
@@ -0,0 +1,42 @@
+from annotypes import Anno, add_call_types
+
+from malcolm.core import PartRegistrar
+from malcolm.modules.builtin.hooks import AContext
+from malcolm.modules.builtin.parts import AMri, APartName, ChildPart
+
+from .. import hooks
+
+with Anno("Whether to raise a ValueError for a bad status"):
+    AErrorOnFail = bool
+
+
+class DirectoryMonitorPart(ChildPart):
+    """Part for checking a directoryMonitor Manager is happy"""
+
+    def __init__(
+        self, name: APartName, mri: AMri, error_on_fail: AErrorOnFail = True
+    ) -> None:
+        super().__init__(name, mri, initial_visibility=True)
+        self.error_on_fail = error_on_fail
+
+    def setup(self, registrar: PartRegistrar) -> None:
+        super().setup(registrar)
+        # Hooks
+        registrar.hook(hooks.ConfigureHook, self.check_directories)
+
+    @add_call_types
+    def check_directories(self, context: AContext) -> None:
+        child = context.block_view(self.mri)
+        # Perform check
+        getattr(child, "managerCheck")()
+
+        # Get status
+        status = getattr(child, "managerStatus").value
```

Comment:
```suggestion
        status = child.managerStatus.value
```
etc.

Reply:
Not even sure what I was thinking here...